joomla-projects/privacy-framework
# [Feature][WIP] Report Privacy Related Capabilities in Admin

### Summary of Changes

This adds a reporting capability to `com_privacy` which allows us to show information to users about privacy related features and capabilities in core and installed extensions.  This screen is meant entirely as an informational dashboard to help site owners have awareness of what integrated extensions are doing.

<img width="1440" alt="screen shot 2018-05-16 at 9 29 00 pm" src="https://user-images.githubusercontent.com/368545/40153630-3dc8ccda-5950-11e8-8063-96694be9db14.png">

This is a read-only screen with append-only capabilities in plugins (meaning one plugin cannot remove the information reported by another and users do not have the ability to remove information in any way except disabling or uninstalling extensions).

### Testing Instructions

Install the base dev branch and apply this patch, or download this branch and install Joomla fresh.  You'll get the new capabilities screen in the component sidebar, navigate to it and you should see a screen with the sliders I have in the screenshot above.  Enabling the reCAPTCHA or language filter plugins should add additional sliders as they are also reporting capabilities.

### Documentation Changes Required

Yep, the plugin integration for this system

### TODO

- [ ] Verify the array merging syntax in the model is actually PHP 5.3 compatible (sorry, I don't have PHP 5 resources available to me anymore so I'm basically relying on CI, code linting, and user testing for this level of backward compatibility)
- [ ] Add reporting for the rest of core, this will be where privacy plugins start getting created
